testsuite/cextras/rcl_arguments.c

```diff
@@ -0,0 +1,3793 @@
+# 1 "testsuite/cextras/rcl_arguments.h"
```

Comment:
3800 line test case? Come on, that's a bit too much.

Reply:
Haha, no it’s still in draft/WIP stage. I’ll shorten it. I created the PR thinking it’d be a small tweak, but I went down a rabbit hole toying with using gcc’s preprocessor. 

On the flip side c2nim now seems to correctly parse the entire preprocessed C source and system headers, including funky Apple stuff. Double function pointers too, ugh.

I still want to use the line directives free to remove all the C includes from ‘cc -E’. That’ll hopefully leave a nice macro expanded C file.